frontend/packages/dev-console/src/actions/add-resources.tsx

```diff
@@ -8,125 +8,138 @@ import {
-  ),
+const resolvedURLWithParams = (
+  unresolvedHref: string,
+  namespace: string,
+  application?: string,
+  contextSource?: string,
+) => {
+  const resolvedURL = resolvedHref(unresolvedHref, namespace);
+  const queryParams = new URLSearchParams();
+  if (application || contextSource) {
+    application
+      ? queryParams.append(QUERY_PROPERTIES.APPLICATION, application)
+      : queryParams.append(QUERY_PROPERTIES.APPLICATION, UNASSIGNED_KEY);
+    contextSource &&
+      queryParams.append(
+        QUERY_PROPERTIES.CONTEXT_ACTION,
+        JSON.stringify({ type: INCONTEXT_ACTIONS_CONNECTS_TO, payload: contextSource }),
+      );
+    return `${resolvedURL}${resolvedURL.indexOf('?') > -1 ? '&' : '?'}${queryParams.toString()}`;
+  }
+  return resolvedURL;
 };
 
-export const containerImage: AddActionItem = {
-  id: 'deploy-image',
-  action: createKebabAction(
-    // t('devconsole~Container Image')
-    'devconsole~Container Image',
-    <OsImageIcon />,
-    ImportOptions.CONTAINER,
-    allCatalogImageResourceAccess,
-  ),
-};
-
-export const fromCatalog: AddActionItem = {
-  id: 'dev-catalog',
-  action: createKebabAction(
-    // t('devconsole~From Catalog')
-    'devconsole~From Catalog',
-    <CatalogIcon />,
-    ImportOptions.CATALOG,
-  ),
-};
-
-export const fromDatabaseCatalog: AddActionItem = {
-  id: 'dev-catalog-databases',
-  action: createKebabAction(
-    // t('devconsole~Database')
-    'devconsole~Database',
-    <DatabaseIcon />,
-    ImportOptions.DATABASE,
-  ),
-};
-
-export const fromSamples: AddActionItem = {
-  id: 'import-from-samples',
-  action: createKebabAction(
-    // t('devconsole~Samples')
-    'devconsole~Samples',
-    <LaptopCodeIcon />,
-    ImportOptions.SAMPLES,
-  ),
-};
-
-export const fromOperatorBacked: AddActionItem = {
-  id: 'operator-backed',
-  action: createKebabAction(
-    // t('devconsole~Operator Backed')
-    'devconsole~Operator Backed',
-    <BoltIcon />,
-    ImportOptions.OPERATORBACKED,
-  ),
-};
-
-export const fromHelmCharts: AddActionItem = {
-  id: 'helm',
-  action: createKebabAction(
-    // t('devconsole~Helm Charts')
-    'devconsole~Helm Charts',
-    <img style={{ height: '1em', width: '1em' }} src={helmIcon} alt="Helm Charts Logo" />,
-    ImportOptions.HELMCHARTS,
-  ),
-};
-
-export const uploadJarFile: AddActionItem = {
-  id: 'upload-jar',
-  action: createKebabAction(
-    // t('devconsole~Upload JAR file')
-    'devconsole~Upload JAR file',
-    <FileUploadIcon />,
-    ImportOptions.UPLOADJAR,
-    allCatalogImageResourceAccess,
-  ),
+export const AddActions: { [name: string]: ActionFactory } = {
+  FromGit: (namespace, application, contextSource, path, accessReviewDisabled) => ({
+    id: 'import-from-git',
+    label: i18next.t('devconsole~Import from Git'),
+    icon: <GitAltIcon />,
+    cta: {
+      href: resolvedURLWithParams('/import/ns/:namespace', namespace, application, contextSource),
+    },
+    path,
+    disabled: !accessReviewDisabled,
```

Comment:
```suggestion
    disabled: accessReviewDisabled,
```

do the same for the other add actions